### DISCUSSION TITLE
Recurring payment ?

### DISCUSSION THREAD

**Discussion by slamer59** [saleor/apps#815] [Integrations & features]
Hello,
I wanted to know if there is a possibility to have recurring payment ? Or payement in X times ? 

One use case is to provide monthly subscription.
Another is to pay for a part of the product when checkout and another time when the product is delivered. 
For exemple, the client pay for 60% the 1st day, we deliver the product and he pays the other 40%.

**Comment by patrys** [comment]
It's not yet supported (at the time of my reply, at least), but we are actively working on it. Programmatic payments won't automate the recurring bit but will provide the necessary APIs to build such a feature as an App. Essentially you will be able to tokenize payment methods at will and then use those tokenized payment methods to create charges in a fully offline manner (offline means neither the customer nor the card itself needs to be present).

You can see the detailed design (and comment on it!) here: https://github.com/saleor/saleor/issues/12843

**Comment by CharlesAttend** [comment]
Hello,
I really love Saleor, it a very impressive apps, I think subscription is the only missing feature. But I can see that your actively working on it 👏
I'm currently working on a project that will need this feature, do you have any idear when it will be out ? 👀
